### PR TITLE
Format src/Compiler/Driver

### PR DESCRIPTION
**Pull request by dsyme**
* [x] Apply formatting to src/Compiler/Driver
* [x] Adjust `fsharp_max_if_then_else_short_width` to be larger (40 --> 60) across all code
* [x] Make a set of pre-formatting changes to make code clearer and avoid horrible code before formatting producing horrible code after formatting

When reviewing this, there effectively two parts 

* The last commit "apply formatting" applies the formatting across the repo
* The 1..N-1 commits either pre-simplify code or adjust the formatting settings

### CODE REVIEW

File: src/Compiler/Driver/CompilerDiagnostics.fs [hunk 1]
```diff
@@ -381,943 +376,1392 @@ let IsWarningOrInfoEnabled (diagnostic, severity) n level specificWarnOn =
...
+        // Strip TargetInvocationException wrappers
+        | :? TargetInvocationException as exn -> SplitRelatedException exn.InnerException
+        | _ -> ToPhased exn, []
 
+    SplitRelatedException diagnostic.Exception
 
 let DeclareMessage = DeclareResourceString
 
 do FSComp.SR.RunStartupValidation()
-let SeeAlsoE() = DeclareResourceString("SeeAlso", "%s")
```
Comment (dsyme) [inline src/Compiler/Driver/CompilerDiagnostics.fs:418]: These changes are unfortunate.  We could increase `fsharp_max_function_binding_width` to make more of them format as single line, but that generally makes other code worse.  

Alternatively we could rid ourselves of all of these and move these strings to FSComp.txt, which doesn't need this insanity

Comment (dsyme) [inline src/Compiler/Driver/CompilerDiagnostics.fs:418]: These changes are unfortunate.  We could increase `fsharp_max_function_binding_width` to make more of them format as single line, but that generally makes other code worse.  

Alternatively we could rid ourselves of all of these and move these strings to FSComp.txt, which doesn't need this insanity

Comment (dsyme) [reply src/Compiler/Driver/CompilerDiagnostics.fs:418]: I've resolved this by increasing `fsharp_max_function_binding_width` for this particular file, it is documented in `.editorconfig`

File: src/Compiler/Driver/CompilerConfig.fs [hunk 2]
```diff
@@ -30,23 +30,22 @@ open FSharp.Compiler.BuildGraph
...
-let mlCompatSuffixes = [".mli";".ml"]
+let FSharpMLCompatFileSuffixes = [ ".mli"; ".ml" ]
 
-let FSharpImplFileSuffixes = [".ml";".fs";".fsscript";".fsx"]
+let FSharpImplFileSuffixes = [ ".ml"; ".fs"; ".fsscript"; ".fsx" ]
 
-let FSharpScriptFileSuffixes = [".fsscript";".fsx"]
+let FSharpScriptFileSuffixes = [ ".fsscript"; ".fsx" ]
 
-let doNotRequireNamespaceOrModuleSuffixes = [".mli";".ml"] @ FSharpScriptFileSuffixes
```
Comment (vzarytovskii) [inline src/Compiler/Driver/CompilerConfig.fs:47]: I presume, that was on purpose, as part of having those in the `FSharpMLCompatFileSuffixes`?

File: src/Compiler/Driver/CompilerDiagnostics.fs [hunk 3]
```diff
@@ -381,943 +376,1082 @@ let IsWarningOrInfoEnabled (diagnostic, severity) n level specificWarnOn =
...
+            | _ -> os.AppendString(ErrorFromAddingTypeEquation1E().Format ty2 ty1 tpcs)
+
+        | ErrorFromAddingTypeEquation (_, _, _, _, (ConstraintSolverTypesNotInEqualityRelation (_, _, _, _, _, contextInfo) as e), _) when
+            (match contextInfo with
+             | ContextInfo.NoContext -> false
+             | _ -> true)
+            ->
+            OutputExceptionR os e
+
+        | ErrorFromAddingTypeEquation (_,
```
Comment (vzarytovskii) [inline src/Compiler/Driver/CompilerDiagnostics.fs:726]: In future, if we use discards, shall we consider putting them on one line?
For example:
```fs
        | ErrorFromAddingTypeEquation (_,_,_,_,
		                                             (ConstraintSolverTypesNotInSubsumptionRelation _
		                                             | ConstraintSolverError _ as e),
		                                             _) -> OutputExceptionR os e
```

**Review by vzarytovskii — APPROVED** [review]

File: src/Compiler/Driver/CompilerDiagnostics.fs (hunk 3 above)
Comment (dsyme) [reply src/Compiler/Driver/CompilerDiagnostics.fs:726]: Possibly yes.  May notice that a lot in CheckExpressions and CheckDeclarations.  

In this case it should probably just be written as two clauses

```fsharp
        | ErrorFromAddingTypeEquation (error=(ConstraintSolverTypesNotInSubsumptionRelation _ as e))
        | ErrorFromAddingTypeEquation (error=(ConstraintSolverError _ as e)) ->
```

File: src/Compiler/Driver/CompilerConfig.fs (hunk 2 above)
Comment (dsyme) [reply src/Compiler/Driver/CompilerConfig.fs:47]: I effectively moved this logic to here: https://github.com/dotnet/fsharp/pull/13195/files#diff-efdad7fa53b23cfadc0c316627cd7f8833d0a531db500cc7c220cc4fa0465ab2R124

**Pull request merged**